kratos/includes/communicator.h

```diff
@@ -623,6 +631,11 @@ class Communicator
         return true;
     }
 
+    virtual bool SynchronizeNonHistoricalVariable(Variable<bool> const& rThisVariable)
```

Comment:
I am just adding the communicator for the bool

Reply:
@adityaghantasala

Reply:
Note that this does not exist for the MPI communicator, so whatever it is that needs this method will not work in MPI.

Reply:
@loumalouomega  considering @jcotela 's comment does it help adding those changes here ? 
I really will prefer to have them in a different PR !

Reply:
I added to MPI too, maybe the implementation is wrong, depending how do we consider the bool works in paralell

Reply:
@loumalouomega does it compile?

Reply:
Indeed, but I will create a separated PR, so you can add feedback directly

Reply:
I have check and I had deactivated the MPI compilation, one sec

Reply:
I have compilation errors with Metis application, I will post an issue